types/git-init/index.d.ts

```diff
@@ -0,0 +1,24 @@
+// Type definitions for git-init 1.0.0
+// Project: https://github.com/yoshuawuyts/git-init
+// Definitions by: Santi <https://github.com/santi100a>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 4.2
+
+import { type ExecException, exec } from "node:child_process";
+
+declare type Callback = (error: ExecException | null, stdout: string, stderr: string) => void;
+
+/**
+ * Initializes a git repo in the current working directory asynchronously.
+ * 
+ * @param cb An {@link exec} compatible callback.
+ */
+declare function init(cb: Callback): void;
+/**
+ * Initializes a git repo asynchronously.
+ * @param path A path to the directory to initialize.
+ * @param cb An {@link exec} compatible callback.
+ */
+declare function init(path: string, cb: Callback): void;
+
+export default init;

```

Comment:
the native module is 8 years old CommonJS file, it cannot be ESM with `default` export:
https://github.com/yoshuawuyts/git-init/blob/master/index.js#L3
please use generator in doubt, also review concept of CJS exports in TS:
https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require